Cython/Compiler/ExprNodes.py

```diff
@@ -520,6 +536,11 @@ def result_as(self, type = None):
             return typecast(type, py_object_type, self.result())
         return typecast(type, self.ctype(), self.result())
 
+    def move_result_as_rhs(self, type = None):
```

Comment:
I found this name confusing in relation with `move_result_rhs()` ("as RHS"? why do we turn the value into a RHS here, and from what?).

I suggest naming it `move_result_rhs_as(type)` and putting it next to `move_result_rhs()`. It's really the same thing, just with a type cast. And then the argument should be mandatory.